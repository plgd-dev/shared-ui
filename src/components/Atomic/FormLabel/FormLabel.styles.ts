import { css } from '@emotion/react'

import { ThemeType, get } from '../_theme'

export const label = (theme: ThemeType) => css`
    font-family: 'Poppins', sans-serif;
    font-size: 12px;
    font-style: normal;
    font-weight: 500;
    line-height: 160%;
    display: inline-block;
    margin-bottom: 10px;
    color: ${get(theme, `FormLabel.color`)};
    cursor: pointer;
`

export const inline = css`
    margin-bottom: 0;
    white-space: nowrap;
    margin-right: 10px;
`

export const w100 = css`
    width: 100%;
`

export const noMargin = css`
    margin-bottom: 0 !important;
`

export const required = (theme: ThemeType) => css`
    color: ${get(theme, `FormLabel.required.color`)};
`

export const flex = css`
    display: flex;
    justify-content: space-between;
    align-items: center;
`
