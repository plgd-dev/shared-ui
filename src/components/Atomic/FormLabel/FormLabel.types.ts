export type Props = {
    className?: string
    id?: string
    inline?: boolean
    required?: boolean
    text: string
    tooltipMaxWidth?: number
    tooltipText?: string
}
