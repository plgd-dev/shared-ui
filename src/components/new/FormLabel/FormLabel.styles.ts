import { css } from '@emotion/react'

export const label = css`
    font-family: 'Poppins', sans-serif;
    font-size: 12px;
    font-style: normal;
    font-weight: 500;
    line-height: 160%;
    display: inline-block;
    margin-bottom: 10px;
    color: #81868c;
    cursor: pointer;
`
