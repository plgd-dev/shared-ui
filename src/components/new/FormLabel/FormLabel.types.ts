export type Props = {
    className?: string
    id?: string
    inline?: boolean
    text: string
}
