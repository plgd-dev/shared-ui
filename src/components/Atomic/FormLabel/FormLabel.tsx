import React, { forwardRef } from 'react'
import { Props } from './FormLabel.types'
import * as styles from './FormLabel.styles'
import ConditionalWrapper from '../ConditionalWrapper'
import { convertSize } from '../Icon'
import Tooltip from '../Tooltip'
import IconQuestion from '../Icon/components/IconQuestion'

const FormLabel = forwardRef<HTMLLabelElement, Props>((props, ref) => {
    const { className, id, inline, marginBottom, required, text, tooltipText, tooltipMaxWidth } = props
    return (
        <label
            className={className}
            css={[styles.label, inline && styles.inline, tooltipText && styles.w100, marginBottom === false && styles.noMargin]}
            htmlFor={id}
            ref={ref}
        >
            <ConditionalWrapper
                condition={!!tooltipText}
                wrapper={(c) => (
                    <span css={styles.flex}>
                        <span>{c}</span>
                        <Tooltip content={tooltipText} delay={200} maxWidth={tooltipMaxWidth} portalTarget={document.getElementById('modal-root')!}>
                            <IconQuestion {...convertSize(16)} />
                        </Tooltip>
                    </span>
                )}
            >
                <>
                    {text}
                    {required && <span css={styles.required}>*</span>}
                </>
            </ConditionalWrapper>
        </label>
    )
})

FormLabel.displayName = 'FormLabel'

export default FormLabel
