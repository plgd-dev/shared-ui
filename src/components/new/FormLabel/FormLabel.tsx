import React, { FC } from 'react'
import { Props } from './FormLabel.types'
import * as styles from './FormLabel.styles'

const FormLabel: FC<Props> = (props) => {
    const { className, id, text } = props
    return (
        <label className={className} css={styles.label} htmlFor={id}>
            {text}
        </label>
    )
}

FormLabel.displayName = 'FormLabel'

export default FormLabel
