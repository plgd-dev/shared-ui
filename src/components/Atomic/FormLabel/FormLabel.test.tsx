import { render } from '../../../../test/jest-wrapper'
import FormLabel from '../FormLabel'

describe('<FormLabel>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <FormLabel text='Placeholder text' />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
