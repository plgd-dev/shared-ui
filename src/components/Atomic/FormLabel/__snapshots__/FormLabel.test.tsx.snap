// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLabel> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Poppins',sans-serif;
  font-size: 12px;
  font-style: normal;
  font-weight: 500;
  line-height: 160%;
  display: inline-block;
  margin-bottom: 10px;
  cursor: pointer;
}

<div>
    <label
      class="emotion-0"
    >
      Placeholder text
    </label>
  </div>
</DocumentFragment>
`;
